src/v2/connectors/redisGraph.js

```diff
@@ -593,9 +593,13 @@ export default class RedisGraphConnector {
     let valuesList = [];
     if (this.rbac.length > 0) {
       const startTime = Date.now();
-      const limitClause = limit <= 0 || property === 'label'
+      let limitClause = limit <= 0 || property === 'label'
         ? ''
         : `LIMIT ${limit}`;
+      if (property === 'image') {
+        // Max image limit of 2500 items, to many returned images causes UI to freeze.
+        limitClause = 'LIMIT 2500';
```

Comment:
Could we use an ENV variable in case we need to ask the customer to use a lower limit?

Reply:
Updated with env var.